docs/en/ingest-management/fleet/add-fleet-server-kubernetes-content.asciidoc

```diff
@@ -0,0 +1,209 @@
++
+[source, shell]
+------------------------------------------------------------
+kubectl create secret generic fleet-server-ssl \
+  --from-file=es-ca.crt=<PATH_TO_ES_CA_CERT_FILE> \
+  --from-file=fleet-ca.crt=<PATH_TO_FLEET_CA_CERT_FILE> \
+  --from-file=fleet-server.crt=<PATH_TO_FLEET_SERVER_CERT> \
+  --from-file=fleet-server.key=<PATH_TO_FLEET_SERVER_CERT_KEY> \
+  --from-literal=fleet_url='<FLEET_URL>'
+------------------------------------------------------------
++
+--
+When running the command, substitute the following values:
+
+* `<PATH_TO_ES_CA_CERT_FILE>` with your local file containing the {es} CA(s).
+* `<PATH_TO_FLEET_CA_CERT_FILE>` with your local file containing the {fleet-server} CA.
+* `<PATH_TO_FLEET_SERVER_CERT>` with your local file containing the server TLS certificate for the {fleet-server}.
+* `<PATH_TO_FLEET_SERVER_CERT_KEY>` with your local file containing the server TLS key for the {fleet-server}.
+* `<FLEET_URL>` with the URL that points to the {fleet-server}, for example `https://fleet-svc`. This URL will be used by the {fleet-server} during its bootstrap, and its hostname must be included in the server certificate’s x509 Subject Alternative Name (SAN) list.
+--
++
+If you prefer to obtain a *yaml manifest* of the Secret to create, append `--dry-run=client -o=yaml` to the command and save the output to a file.
+// end::production-secret[]
+
+// ***************************************************
+// ***************************************************
+
+// tag::quickstart-deployment[]
+["source","yaml",subs="attributes"]
+------------------------------------------------------------
+apiVersion: v1
+kind: Service
+metadata:
+  name: fleet-svc
+spec:
+  type: ClusterIP
+  selector:
+    app: fleet-server
+  ports:
+  - port: 443
+    protocol: TCP
+    targetPort: 8220
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: fleet-server
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: fleet-server
+  template:
+    metadata:
+      labels:
+        app: fleet-server
+    spec:
+      automountServiceAccountToken: false
+      containers:
+      - name: elastic-agent
+        image: docker.elastic.co/beats/elastic-agent:{version}
+        env:
+          - name: FLEET_SERVER_ENABLE
+            value: "true"
+          - name: FLEET_SERVER_ELASTICSEARCH_HOST
+            valueFrom:
+              secretKeyRef:
+                name: fleet-server-config
+                key: elastic_endpoint
+          - name: FLEET_SERVER_SERVICE_TOKEN
+            valueFrom:
+              secretKeyRef:
+                name: fleet-server-config
+                key: elastic_service_token
+          - name: FLEET_SERVER_POLICY_ID
+            valueFrom:
+              secretKeyRef:
+                name: fleet-server-config
+                key: fleet_policy_id
+          - name: ELASTICSEARCH_CA
```

Comment:
qq @eedugon is this actually needed for the quick-start deployment? 🙂

Reply:
> is this actually needed for the quick-start deployment?
@pkoutsovasilis , thanks a lot for looking into this!

Yes, I think the `ELASTICSEARCH_CA` is needed in both quick start and production modes / cases.

Quick start mode only affects the `HTTPS server` part of the `fleet server`. In quickstart mode the fleet server creates a `self-signed` certificate and the clients need to be enrolled with `--insecure` flag, but that doesn't affect the traffic flows with Elasticsearch.

At least that's how I understand this, and in my tests I always need to set the ELASTICSEARCH_CA of my self-managed elasticsearch cluster when booting a Fleet Server in quickstart mode.

Reply:
well, of course if the Elasticsearch certificate is signed by a well known and public CA then `ELASTICSEARCH_CA` won't be needed, but that will also be the same no matter `quick start` / `production`.

I believe I have also mentioned that in the doc in a couple of places.